ambari-logsearch/ambari-logsearch-web/src/app/services/logs-container.service.ts

```diff
@@ -885,32 +883,35 @@ export class LogsContainerService {
   }
 
   openServiceLog(log: ServiceLog): void {
-    const tab = {
-      id: log.id || `${log.host}-${log.type}`,
-      isCloseable: true,
-      label: `${log.host} >> ${log.type}`,
-      activeFilters: Object.assign(this.getFiltersData('serviceLogs'), {
-        components: this.filters.components.options.find((option: ListItem): boolean => {
-          return option.value === log.type;
-        }),
-        hosts: this.filters.hosts.options.find((option: ListItem): boolean => {
-          return option.value === log.host;
-        })
-      }),
-      appState: {
-        activeLogsType: 'serviceLogs',
-        isServiceLogsFileView: true,
-        activeLog: {
-          id: log.id,
-          host_name: log.host,
-          component_name: log.type
-        }
-      }
-    };
-    this.tabsStorage.addInstance(tab);
-    this.router.navigate(['/logs', tab.id], {
-      queryParams: this.logsFilteringUtilsService.getQueryParamsFromActiveFilter(tab.activeFilters, 'serviceLogs')
-    });
-  }
-
+    this.componentsService.findInCollection(component => (component.name || component.label) === log.type)
+      .map(component => component ? component.label || component.name : name)
+      .first()
+      .subscribe((componentName) => {
+        const tab = {
+          id: log.id || `${log.host}-${log.type}`,
+          isCloseable: true,
+          isActive: false,
+          label: `${log.host} >> ${componentName || log.type}`,
+          activeFilters: Object.assign({}, JSON.parse(JSON.stringify(this.filtersForm.value)), {
```

Comment:
What's `JSON.parse(JSON.stringify(this.filtersForm.value))` for?

Reply:
Poor man deep copy :)
It is to cut off all the possibly references.